### PR TITLE
chore(*) get rid of advertised hostname

### PR DESCRIPTION
**Pull request by jakubdyszkiewicz**
### Summary

The goal was to get rid of `kuma_general_advertised_hostname`. Overall it was confusing for the users that it needs to be set.

It was used in a couple of use cases

### Catalog
All binaries queries API Server and get back the URL/address of proper server.
Although it was nice, CP needed to know about it's own deployment topology which is weird and should be out of the scope of the application itself.
This also required to open catalog and API Server to all the clients. Kuma DP already is changed so it uses proper port for CP-DP communication.
I changed the Kuma Prometheus SD to also follow the same pattern (connect on the port of the MADS server, not on catalog).
Catalog should be gone in the next major version of Kuma. We keep it for now, because it will smooth out migration to 1.0

### CP points back DP to the CP
In some cases CP had to point back DP to the CP:
Bootstrap configuration that is pointing back to XDS XDS that is pointing back to SDS.
I changed it so we use Host/:authority header this way we know on which hostname DP wants to talk with CP and we only change ports.

### Generating certificates
On Universal, we autogenerate TLS certificates with SAN of KUMA_GENERAL_ADVERTISED_HOSTNAME. Those certs are then used for securing all the servers in Kuma.
When Envoy connects to the CP using XDS, it uses this cert and verifies that the SAN in cert = hostname that it tries to connect.

We could skip SAN verification on Bootstrap endpoint of DP server, but unfortunately, we cannot skip SAN verification on XDS/SDS of DP server
https://www.envoyproxy.io/docs/envoy/latest/api-v2/api/v2/core/grpc_service.proto#envoy-api-msg-core-grpcservice-googlegrpc-sslcredentials

The usefulness of autogenerated certs on production is questionable because they are not synced between instances.

Now we are generating certs with all IPs and hostname from all network interfaces and we print log what IPs/hostnames are available.
So it will work if you are playing with Kuma on a couple of machines, but won't if you put it behind loadbalancer unless you generate proper cert (which you have to do on prod anyways).

On K8s this is automated (we know the hostname when generating installation from HELM chart).

### Docs

[ ] - todo change MADS URL

### CODE REVIEW

**Review by nickolaev — APPROVED** [review]
Only a remark on the changed test

File: pkg/api-server/pagination.go [hunk 1]
```diff
@@ -40,16 +38,21 @@ func pagination(request *restful.Request) (page, error) {
 	}, nil
 }
 
-func nextLink(request *restful.Request, publicURL string, list model.ResourceList) (*string, error) {
+func nextLink(request *restful.Request, list model.ResourceList) (*string, error) {
```
Comment (lobkovilya) [inline pkg/api-server/pagination.go:41]: Seems like we don't need to return an error in this method

File: pkg/catalog/types.go [hunk 2]
```diff
@@ -9,9 +9,8 @@ type Catalog struct {
 }
 
 type Apis struct {
-	Bootstrap            BootstrapApi            `json:"bootstrap"`
-	DataplaneToken       DataplaneTokenApi       `json:"dataplaneToken"` // DEPRECATED: remove in next major version of Kuma
-	MonitoringAssignment MonitoringAssignmentApi `json:"monitoringAssignment"`
+	Bootstrap      BootstrapApi      `json:"bootstrap"`
+	DataplaneToken DataplaneTokenApi `json:"dataplaneToken"` // DEPRECATED: remove in next major version of Kuma
```
Comment (lobkovilya) [inline pkg/catalog/types.go:13]: Is it the right time to remove it?

File: pkg/config/api-server/catalog/config.go [hunk 3]
```diff
@@ -2,15 +2,8 @@ package catalog
...
-	ApiServer            ApiServerConfig               `yaml:"-"`
-	Bootstrap            BootstrapApiConfig            `yaml:"bootstrap"` // DEPRECATED: remove in next major version of Kuma
-	DataplaneToken       DataplaneTokenApiConfig       `yaml:"-"`         // DEPRECATED: remove in next major version of Kuma
-	MonitoringAssignment MonitoringAssignmentApiConfig `yaml:"monitoringAssignment"`
-	Sds                  SdsApiConfig                  `yaml:"sds"` // DEPRECATED: remove in next major version of Kuma
-}
-
-type ApiServerConfig struct {
-	Url string
+	Bootstrap      BootstrapApiConfig      `yaml:"bootstrap"` // DEPRECATED: remove in next major version of Kuma
```
Comment (lobkovilya) [inline pkg/config/api-server/catalog/config.go:5]: And this?

**Review by lobkovilya — APPROVED** [review]

File: pkg/catalog/types.go (hunk 2 above)
Comment (jakubdyszkiewicz) [reply pkg/catalog/types.go:13]: It's not :( If we remove it, Kuma DP 0.x will fail to connect to 1.0 CP

File: pkg/config/api-server/catalog/config.go (hunk 3 above)
Comment (jakubdyszkiewicz) [reply pkg/config/api-server/catalog/config.go:5]: If we remove it, Kuma DP 0.x will fail to connect to 1.0 CP

**Pull request merged**